PROTO_tests/tests/io_test.py

```diff
@@ -491,3 +525,735 @@ def test_ipv4_columns(self, comp):
         df = ak.DataFrame({"a": ak.IPv4(ak.arange(10))})
         df["a"] = df["a"].export_uint()
         assert ak.arange(10).to_list() == df["a"].to_list()
+
+
+class TestHDF5:
+    hdf_test_base_tmp = f"{os.getcwd()}/hdf_io_test"
+    io_util.get_directory(hdf_test_base_tmp)
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NUMERIC_AND_STR_TYPES)
+    def test_read_and_write(self, prob_size, dtype):
+        ak_arr = make_ak_arrays(prob_size * pytest.nl, dtype)
+        with tempfile.TemporaryDirectory(dir=TestHDF5.hdf_test_base_tmp) as tmp_dirname:
+            file_name = f"{tmp_dirname}/hdf_test_correct"
+            ak_arr.to_hdf(file_name, "my_dset")
+
+            # test read_hdf with glob
+            gen_arr = ak.read_hdf(f"{file_name}*", "my_dset")
+            assert (ak_arr == gen_arr).all()
+
+            # test read_hdf with filenames
+            gen_arr = ak.read_hdf(filenames=[f"{file_name}_LOCALE{i:04d}" for i in range(pytest.nl)])
```

Comment:
This is the only case where we do not provide the dataset name. I would recommend that we test both cases for everything. This way we ensure all the workflows are operating as expected. 

I am thinking it may be easiest to have a `test_read_and_write` and `test_read_and_write_dset_provided`. That way we are fully testing each case. 

We may also want to test when a list of datasets is specified with multiple dsets. Again, testing that `ls` works properly when not supplied, a subset can be selected properly and all can be specified by the user. This may be overkill, but I think it would be beneficial.

Reply:
I believe this should be updated. We now have both `test_read_and_write` and `test_read_and_write_dset_provided`. 

The request to test reading from a file with multiple dataset and specify no datasets, 1 dataset, half datasets, and all datasets was added to `test_read_and_write_with_dict`. 

Let me know if I misunderstood what you were asking for